lib/stat_tracker.rb

```diff
@@ -56,13 +56,82 @@ def percentage_visitor_wins
     (away_win?.to_f / @games.size).round(2)
   end
 
+  def percentage_ties
+    tie_games = @game_teams.find_all do |game|
+      game.game_result.include?("TIE")
+    end
+    results = tie_games.length.to_f / @game_teams.length.to_f
+    results.round(2)
+  end
+
+  def count_of_games_by_season
+    ids = []
+    only_ids = []
+      @games.find_all do |game|
+        ids = game.season_id
+        only_ids << ids
+      end
+    uniq_ids = only_ids.uniq
+    count_games = []
+    final_count = Hash.new
+      uniq_ids.each do |id|
+        count_games = @games.find_all {|game|game.season_id.include?(id)}
+        final_count[id] = count_games.length
+      end
+      final_count
+  end
+
   # LEAGUE STATS
 
   def count_of_teams
     @teams.count
   end
 
+  def highest_scoring_home_team
```

Comment:
I believe you could use a helper method like this to avoid duplicating the calculation of highest scoring and lowest scoring teams.

`home_teams_goals = Hash.new { |hash, key| hash[key] = [] }`

`@game_teams.each do |game_team|`
`  home_teams_goals[game_team.team_id.to_sym] << game_team.goals.to_i if game_team.home_or_away == "home"`
`end`

Reply:
I really like this idea, Mike! It could be used in some methods I wrote as well.